pkg/deploy/controller/deploymentcancellation/controller_test.go

```diff
@@ -0,0 +1,166 @@
+package deploymentcancellation
+
+import (
+	"testing"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/record"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deploytest "github.com/openshift/origin/pkg/deploy/api/test"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+)
+
+// TestHandle_cancelInflightDeployment ensures that a deployment in the
+// new/pending/running state can be cancelled
+// and a deployment in failed/complete state cannot be cancelled
+func TestHandle_cancelInflightDeployment(t *testing.T) {
+	var updatedPod *kapi.Pod
+	deployment := cancelledDeployment()
+
+	controller := &DeploymentCancellationController{
+		podClient: &podClientImpl{
```

Comment:
```FakePods``` would work just as well here.  Have you seen ```NewSimpleFake```?  We end up with a lot of code on the page that does alot of the same things as what upstream has.